### PR TITLE
fix shell checks errors in build/lib/release.sh

### PR DESCRIPTION
**Pull request by beautytiger**
**What type of PR is this?**
/kind bug
/kind cleanup

**What this PR does / why we need it**:
makes `./build/lib/release.sh` shell scripts pass hack/verify-shellcheck.sh
**Which issue(s) this PR fixes**:

ref: #72956

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
NONE
```

### CODE REVIEW

File: build/lib/release.sh [hunk 1]
```diff
@@ -106,14 +106,14 @@ function kube::release::package_src_tarball() {
...
-      $(cd "${KUBE_ROOT}" && find . -mindepth 1 -maxdepth 1 \
+      "$(cd "${KUBE_ROOT}" && find . -mindepth 1 -maxdepth 1 \
         -not \( \
           \( -path ./_\*        -o \
              -path ./.git\*     -o \
              -path ./.config\* -o \
              -path ./.gsutil\*    \
           \) -prune \
-        \))
+        \))"
```
Comment (k-toyoda-pi) [inline build/lib/release.sh]: original code expects source_files to be an array.
If double quote is added, it becomes single element. So the result is different from original. How about using `while` and `read` according to https://github.com/koalaman/shellcheck/wiki/SC2207?

Comment (beautytiger) [reply build/lib/release.sh]: Good point!
Thank you very much.
Changed this according to your suggestion.

File: build/lib/release.sh [hunk 2]
```diff
@@ -37,39 +37,6 @@ KUBE_BUILD_PULL_LATEST_IMAGES=${KUBE_BUILD_PULL_LATEST_IMAGES:-y}
...
-#   If version is a valid ci version
-# Sets:                    (e.g. for '1.2.3-alpha.4.56+abcdef12345678')
-#   VERSION_MAJOR          (e.g. '1')
-#   VERSION_MINOR          (e.g. '2')
-#   VERSION_PATCH          (e.g. '3')
-#   VERSION_PRERELEASE     (e.g. 'alpha')
-#   VERSION_PRERELEASE_REV (e.g. '4')
-#   VERSION_BUILD_INFO     (e.g. '.56+abcdef12345678')
-#   VERSION_COMMITS        (e.g. '56')
-function kube::release::parse_and_validate_ci_version() {
```
Comment (beautytiger) [inline build/lib/release.sh]: Nowhere else consume this function, I think it's time to say goodbye.
is this right? @BenTheElder

Comment (BenTheElder) [reply build/lib/release.sh]: did we check other repos? http://cs.k8s.io  can help with that

File: build/lib/release.sh [hunk 3]
```diff
@@ -449,12 +419,12 @@ function kube::release::package_kube_manifests_tarball() {
...
   local objects
   objects=$(cd "${KUBE_ROOT}/cluster/addons" && find . \( -name \*.yaml -or -name \*.yaml.in -or -name \*.json \) | grep -v demo)
-  tar c -C "${KUBE_ROOT}/cluster/addons" ${objects} | tar x -C "${dst_dir}"
+  tar c -C "${KUBE_ROOT}/cluster/addons" "${objects}" | tar x -C "${dst_dir}"
   # Merge GCE-specific addons with general purpose addons.
   local gce_objects
   gce_objects=$(cd "${KUBE_ROOT}/cluster/gce/addons" && find . \( -name \*.yaml -or -name \*.yaml.in -or -name \*.json \) \( -not -name \*demo\* \))
   if [[ -n "${gce_objects}" ]]; then
-    tar c -C "${KUBE_ROOT}/cluster/gce/addons" ${gce_objects} | tar x -C "${dst_dir}"
+    tar c -C "${KUBE_ROOT}/cluster/gce/addons" "${gce_objects}" | tar x -C "${dst_dir}"
```
Comment (BenTheElder) [inline build/lib/release.sh]: what format is gce_objects? were we depending on space splitting?

File: build/lib/release.sh [hunk 4]
```diff
@@ -449,12 +419,12 @@ function kube::release::package_kube_manifests_tarball() {
   cp "${KUBE_ROOT}/cluster/gce/gci/health-monitor.sh" "${dst_dir}/health-monitor.sh"
   local objects
   objects=$(cd "${KUBE_ROOT}/cluster/addons" && find . \( -name \*.yaml -or -name \*.yaml.in -or -name \*.json \) | grep -v demo)
-  tar c -C "${KUBE_ROOT}/cluster/addons" ${objects} | tar x -C "${dst_dir}"
+  tar c -C "${KUBE_ROOT}/cluster/addons" "${objects}" | tar x -C "${dst_dir}"
```
Comment (BenTheElder) [inline build/lib/release.sh]: what format is objects? were we depending on space splitting?

File: build/lib/release.sh [hunk 5]
```diff
@@ -349,14 +320,13 @@ function kube::release::create_docker_images_for_server() {
 
       local oldifs=$IFS
       IFS=","
-      set $wrappable
+      set "$wrappable"
```
Comment (BenTheElder) [inline build/lib/release.sh]: hmm what is this used for? what are the contents?

File: build/lib/release.sh [hunk 6]
```diff
@@ -333,7 +304,7 @@ function kube::release::create_docker_images_for_server() {
     local binary_dir="$1"
     local arch="$2"
     local binary_name
-    local binaries=($(kube::build::get_docker_wrapped_binaries "${arch}"))
+    local binaries=("$(kube::build::get_docker_wrapped_binaries "${arch}")")
```
Comment (BenTheElder) [inline build/lib/release.sh]: I think we were intentionally splitting here? were we?

File: build/lib/release.sh (hunk 2 above)
Comment (beautytiger) [reply build/lib/release.sh]: Got it. I have checked, no real use of this function across all repos.
Ready to delete ;)

File: build/lib/release.sh (hunk 6 above)
Comment (beautytiger) [reply build/lib/release.sh]: Yes, it is an array. I will modify again.

File: build/lib/release.sh (hunk 5 above)
Comment (beautytiger) [reply build/lib/release.sh]: wrappable is like this:
```
kube-apiserver,k8s.io/debian-base-amd64:v1.0.0
```
it is comma separated, this change is wrong, I will suppress it.

File: build/lib/release.sh (hunk 3 above)
Comment (beautytiger) [reply build/lib/release.sh]: Yes, space splitting is required, so I just suppress this check here.
Thanks

File: build/lib/release.sh (hunk 4 above)
Comment (beautytiger) [reply build/lib/release.sh]: Yes, After test, I found space splitting is required, so I just suppress this check here.
Thanks!

File: build/lib/release.sh [hunk 7]
```diff
@@ -333,7 +304,8 @@ function kube::release::create_docker_images_for_server() {
     local binary_dir="$1"
     local arch="$2"
     local binary_name
-    local binaries=($(kube::build::get_docker_wrapped_binaries "${arch}"))
+    local binaries=()
+    IFS=" " read -r -a binaries <<< "$(kube::build::get_docker_wrapped_binaries "${arch}")"
```
Comment (BenTheElder) [inline build/lib/release.sh]: @cblecker do we want this or the util method?

File: build/lib/release.sh [hunk 8]
```diff
@@ -124,8 +91,10 @@ function kube::release::package_src_tarball() {
 function kube::release::package_client_tarballs() {
    # Find all of the built client binaries
   local platform platforms
-  platforms=($(cd "${LOCAL_OUTPUT_BINPATH}" ; echo */*))
+  platforms=()
+  IFS=" " read -r -a platforms <<< "$(cd "${LOCAL_OUTPUT_BINPATH}" || exit 2; echo ./*/*)"
```
Comment (beautytiger) [inline build/lib/release.sh]: +1

**Comment by BenTheElder** [comment]
cc @justaugustus :+)

**Comment by BenTheElder** [comment]
I think this is about right, will do another pass sometime during the day tomorrow.

File: build/lib/release.sh [hunk 9]
```diff
@@ -419,11 +396,15 @@ function kube::release::package_kube_manifests_tarball() {
   cp "${KUBE_ROOT}/cluster/gce/gci/health-monitor.sh" "${dst_dir}/health-monitor.sh"
   local objects
   objects=$(cd "${KUBE_ROOT}/cluster/addons" && find . \( -name \*.yaml -or -name \*.yaml.in -or -name \*.json \) | grep -v demo)
+  # word splitting by space is intentional and required so disable checkcheck for this line
+  # shellcheck disable=SC2086
   tar c -C "${KUBE_ROOT}/cluster/addons" ${objects} | tar x -C "${dst_dir}"
```
Comment (BenTheElder) [inline build/lib/release.sh]: how much effort is involved in turning these into arrays? (didn't trace the source)

Comment (beautytiger) [reply build/lib/release.sh]: Oh. Thanks for your suggestion. There are ways to change it into array and it works.

File: build/lib/release.sh (hunk 7 above)
Comment (BenTheElder) [reply build/lib/release.sh]: ping :-)

**Comment by spiffxp** [comment]
/hold
I don't want this to merge until https://github.com/kubernetes/kubernetes/issues/86182 has been root-caused

**Comment by beautytiger** [comment]
Too many conflicts in rebase, I will close this and redo it later.

**Pull request closed without merging**